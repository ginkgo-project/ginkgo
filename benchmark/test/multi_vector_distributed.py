#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output_distributed(
    ["-input", '[{"n": 100}]'],
    expected_stdout="multi_vector_distributed.simple.stdout",
    expected_stderr="multi_vector_distributed.simple.stderr",
    num_procs=3,
)

# stdin
test_framework.compare_output_distributed(
    [],
    expected_stdout="multi_vector_distributed.simple.stdout",
    expected_stderr="multi_vector_distributed.simple.stderr",
    stdin='[{"n": 100}]',
    num_procs=3,
)

# file
test_framework.compare_output_distributed(
    ["-input", str(test_framework.sourcepath / "input.blas.json")],
    expected_stdout="multi_vector_distributed.simple.stdout",
    expected_stderr="multi_vector_distributed.simple.stderr",
    num_procs=3,
)

# profiler annotations
test_framework.compare_output_distributed(
    ["-input", '[{"n": 100}]', "-profile", "-profiler_hook", "debug"],
    expected_stdout="multi_vector_distributed.profile.stdout",
    expected_stderr="multi_vector_distributed.profile.stderr",
    num_procs=3,
)

# complex
test_framework.compare_output_distributed(
    ["-input", '[{"n": 100}]'],
    expected_stdout="multi_vector_distributed_dcomplex.simple.stdout",
    expected_stderr="multi_vector_distributed_dcomplex.simple.stderr",
    num_procs=3,
    use_complex=True
)
