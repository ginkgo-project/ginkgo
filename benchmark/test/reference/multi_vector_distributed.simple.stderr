Running on reference(0)
Running with 2 warm iterations and 10 running iterations
The random seed for right hand sides is 42
The operations are copy,axpy,scal
Running test case n = 100 
	Running blas: copy
	Running blas: axpy
	Running blas: scal
